x-pack/plugins/reporting/server/export_types/csv_saved_object/execute_job.ts

```diff
@@ -0,0 +1,109 @@
+import type { VisualizationSavedObjectAttributes } from 'src/plugins/visualizations/common';
+import { DeepPartial } from 'utility-types';
+import { JobParamsCSV } from '../..';
+import { injectReferences, parseSearchSourceJSON } from '../../../../../../src/plugins/data/common';
+import { CSV_JOB_TYPE } from '../../../common/constants';
+import { getFieldFormats } from '../../services';
+import type { RunTaskFn, RunTaskFnFactory } from '../../types';
+import { decryptJobHeaders } from '../common';
+import { CsvGenerator } from '../csv_searchsource/generate_csv';
+import { getSharingData } from './lib';
+import type { TaskPayloadCsvFromSavedObject } from './types';
+
+type RunTaskFnType = RunTaskFn<TaskPayloadCsvFromSavedObject>;
+type SavedSearchObjectType = SavedObject<
+  VisualizationSavedObjectAttributes & { columns?: string[]; sort: Array<[string, string]> }
+>;
+type ParsedSearchSourceJSON = SearchSourceFields & { indexRefName?: string };
+
+function isSavedObject(
+  savedSearch: SavedSearchObjectType | unknown
+): savedSearch is SavedSearchObjectType {
+  return (
+    (savedSearch as DeepPartial<SavedSearchObjectType> | undefined)?.attributes
+      ?.kibanaSavedObjectMeta?.searchSourceJSON != null
+  );
+}
+
+export const runTaskFnFactory: RunTaskFnFactory<RunTaskFnType> = (reporting, _logger) => {
+  const config = reporting.getConfig();
+
+  return async function runTask(jobId, job, cancellationToken, stream) {
+    const logger = _logger.clone([CSV_JOB_TYPE, 'execute-job', jobId]);
+
+    const encryptionKey = config.get('encryptionKey');
+    const headers = await decryptJobHeaders(encryptionKey, job.headers, logger);
+    const fakeRequest = reporting.getFakeRequest({ headers }, job.spaceId, logger);
+    const uiSettings = await reporting.getUiSettingsClient(fakeRequest, logger);
+    const savedObjects = await reporting.getSavedObjectsClient(fakeRequest);
+    const dataPluginStart = await reporting.getDataService();
+    const fieldFormatsRegistry = await getFieldFormats().fieldFormatServiceFactory(uiSettings);
+
+    const [es, searchSourceStart] = await Promise.all([
+      (await reporting.getEsClient()).asScoped(fakeRequest),
+      await dataPluginStart.search.searchSource.asScoped(fakeRequest),
+    ]);
+
+    const clients = {
+      uiSettings,
+      data: dataPluginStart.search.asScoped(fakeRequest),
+      es,
+    };
+    const dependencies = {
+      searchSourceStart,
+      fieldFormatsRegistry,
+    };
+
+    // Get the Saved Search Fields object from ID
+    const savedSearch = await savedObjects.get('search', job.savedObjectId);
+
+    if (!isSavedObject(savedSearch)) {
+      throw new Error(`Saved search object is not valid`);
+    }
+
+    // allowed to throw an Invalid JSON error if the JSON is not parseable.
+    const searchSourceFields: ParsedSearchSourceJSON = parseSearchSourceJSON(
+      savedSearch.attributes.kibanaSavedObjectMeta.searchSourceJSON
+    );
```

Comment:
Think a bit ahead here. As I understand this team "owns" the "search source" object but not the "search" object?

When we have support for "down" and "on-read" migrations of saved objects we can consider pinning this object to a specific version of "search", but also a specific version of "searchsource" to ensure it is always in the shape we expect.

This is nothing to action on this PR, I'd be interested to hear what @pgayvallet or @rudolf think :)